src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.Impl.cs

```diff
@@ -20,69 +20,23 @@ namespace Microsoft.CodeAnalysis.PublicApiAnalyzers
 {
     public partial class DeclarePublicApiAnalyzer : DiagnosticAnalyzer
     {
-        private sealed class ApiLine
-        {
-            public string Text { get; }
-            public TextSpan Span { get; }
-            public SourceText SourceText { get; }
-            public string Path { get; }
-            public bool IsShippedApi { get; }
-
-            internal ApiLine(string text, TextSpan span, SourceText sourceText, string path, bool isShippedApi)
-            {
-                Text = text;
-                Span = span;
-                SourceText = sourceText;
-                Path = path;
-                IsShippedApi = isShippedApi;
-            }
-        }
+        private sealed record AdditionalFileInfo(string Path, SourceText SourceText, bool IsShippedApi);
 
-#pragma warning disable CA1815 // Override equals and operator equals on value types
-        private readonly struct RemovedApiLine
-#pragma warning restore CA1815 // Override equals and operator equals on value types
+        private readonly record struct ApiLine(string Text, TextSpan Span, AdditionalFileInfo FileInfo)
         {
-            public string Text { get; }
-            public ApiLine ApiLine { get; }
-
-            internal RemovedApiLine(string text, ApiLine apiLine)
-            {
-                Text = text;
-                ApiLine = apiLine;
-            }
+            public SourceText SourceText => FileInfo.SourceText;
+            public string Path => FileInfo.Path;
+            public bool IsShippedApi => FileInfo?.IsShippedApi is true;
```

Comment:
>FileInfo? [](http://example.com/codeflow?start=40&length=9)

why null check?

Reply:
It's a struct, and used in an fashion where it can be in the default state.

Reply:
So should the other ones be checked then too?